src/universal-dapp.js

```diff
@@ -679,34 +680,94 @@ function tryTillResponse (web3, txhash, done) {
 
 UniversalDApp.prototype.runTx = function (args, cb) {
   var self = this
-  var to = args.to
-  var data = args.data
-  if (data.slice(0, 2) !== '0x') {
-    data = '0x' + data
+  var tx = {
+    to: args.to,
+    data: args.data
   }
 
-  var gasLimit = 3000000
-  if (self.getGasLimit) {
-    try {
-      gasLimit = self.getGasLimit()
-    } catch (e) {
-      return cb(e)
-    }
-  }
+  async.waterfall([
+    // query gas limit
+    function (callback) {
```

Comment:
Is that possible to move these functions (also getGasLimit maybe) outside of universal-dapp.js?
cause if we add more functionality here, the whole universal-dapp.js will become not so readable.